coops-ui/src/components/AnnualReport/Directors.vue

```diff
@@ -598,7 +609,21 @@ export default class Directors extends Mixins(DateMixin, ExternalMixin, EntityFi
   private get directorFirstNameRules (): Array<Function> {
     return [
       v => !!v || 'A first name is required',
-      v => !/\s/g.test(v) || 'Invalid spaces'
+      v => !/^\s/g.test(v) || 'Invalid spaces', // leading spaces
+      v => !/\s$/g.test(v) || 'Invalid spaces', // trailing spaces
+      v => !/\s\s/g.test(v) || 'Invalid word spacing' // multiple inline spaces
+    ]
+  }
+
+  /**
+   * Computed value.
+   * @returns The array of validations rules for a director's middle initial.
+   */
+  private get directorMiddleInitialRules (): Array<Function> {
+    return [
+      v => !/^\s/g.test(v) || 'Invalid spaces', // leading spaces
+      v => !/\s$/g.test(v) || 'Invalid spaces', // trailing spaces
+      v => !/\s\s/g.test(v) || 'Invalid word spacing' // multiple inline spaces
```

Comment:
Can we create one ruleMethod() and apply them to all of the fields where required, it appears that all of the error prompts are similar. ie..   

`private get directorNameRules (nameValidated: string): Array<Function> {
    return [
      v => !!v || `A ${nameValidated} name is required`,
      v => !/\s/g.test(v) || 'Invalid spaces'
      v => !/^\s/g.test(v) || 'Invalid spaces', // leading spaces
      v => !/\s$/g.test(v) || 'Invalid spaces', // trailing spaces
      v => !/\s\s/g.test(v) || 'Invalid word spacing' // multiple inline spaces
    ]
  }

Reply:
Yes, I could write a property that is an array of the common rules, and spread it within each first/middle/last name getter.
I suggest that I (or another dev) do that the next time this code is touched or rules are added. OK?

Reply:
100%, fine by me!

Reply:
FYI - please check in postgres (prod, ideally) whether there are director names that break these rules (leading spaces, trailing spaces, multiple spaces between words). If so, we should either fix the data, or not be so strict. Otherwise users will get errors without actually changing their data.

Reply:
I checked in Prod, and yes there are some director names with trailing / multiple inline spaces.

I entered /bcgov/entity#1968 to update the Prod DB and Legal Updater accordingly.

Reply:
Not being strict is another possibility, ie, don't allow only spaces, but allow leading / trailing / multiple inline spaces. But I would prefer to see sanitized data.

Reply:
@katiemcgoff Do we have a decision on whether to allow spaces in a field (but not only spaces)?